ApplicationInsights/JavaScriptSDK/Telemetry/PageViewManager.ts

```diff
@@ -44,74 +47,82 @@ export class PageViewManager {
     *
     * In all cases page view performance is sent once (only for the 1st call of trackPageView), or not sent if navigation timing is not supported.
     */
-    public trackPageView(name?: string, url?: string, properties?: Object, measurements?: Object, duration?: number) {
-        // ensure we have valid values for the required fields
-        if (typeof name !== "string") {
-            name = window.document && window.document.title || "";
+    public trackPageView(pageView: IPageViewTelemetry, customProperties?: { [key: string]: any }) {
+        let name = pageView.name;
+        if (CoreUtils.isNullOrUndefined(name) || typeof name !== "string") {
+            pageView.name = window.document && window.document.title || "";
         }
 
-        if (typeof url !== "string") {
-            url = window.location && window.location.href || "";
+        let uri = pageView.uri;
+        if (CoreUtils.isNullOrUndefined(uri) || typeof uri !== "string") {
+            pageView.uri = window.location && window.location.href || "";
         }
 
-        var pageViewSent = false;
-        var customDuration = undefined;
-
-        if (PageViewPerformance.isPerformanceTimingSupported()) {
```

Comment:
so, this part we can restore to the original caclulation because it is calculating based on browser capability. Restore this entire if loop.

Reply:
discussed with Piyali offline. code logic did not change. We traced the code to ensure scenarios are still covered.

---
In reply to: [205504455](https://github.com/Microsoft/ApplicationInsights-JS/pull/635#discussion_r205504455) [](ancestors = 205504455)